src/app/app.component.ts

```diff
@@ -164,9 +167,18 @@ export class AppComponent {
       this.footerDefaults = this.appConfig.APP_FOOTER_DEFAULTS;
       this.appAuthenticationDefaults = this.appConfig.APP_AUTHENTICATION_DEFAULTS;
       this.appFields = this.appConfig.APP_FIELDS;
+      this.shouldCollapseHeader = this.appConfig.APP_HEADER_DEFAULTS.collapsing;
     });
   }
 
+  /** Listen to scroll events to trigger header collapse */
+  @HostListener("ionScroll", ["$event"])
+  scrolling($event: any) {
+    if (this.shouldCollapseHeader) {
```

Comment:
nit(non-blocking) - this will be adding event listeners even when not using collapseHeader. Is there a way to conditionally add the listener instead of conditionally forwarding each event?